modules/mod_ginger_base/controllers/controller_rest_resources.erl

```diff
@@ -225,15 +228,67 @@ trans({Key, Value}, Acc) ->
 supported_search_args() ->
     ["cat", "hasobject", "hassubject", "sort"].
 
-rsc(Id, Context, IncludeEdges) ->
-    Map = m_ginger_rest:rsc(Id, Context),
-    case IncludeEdges of
-        false ->
-            Map;
-        true ->
-            m_ginger_rest:with_edges(Map, Context)
+parse_path(String) ->
+    case erl_scan:string(String) of
+        {ok, Tokens, _} ->
+            case erl_parse:parse_term(Tokens ++ [{dot, 1}]) of
+                            {ok, Path} when is_list(Path) -> Path;
+                            _ -> [[]]
+                        end;
+        _ -> [[]]
     end.
 
+%% @doc Add edges to resource based on depth path
+%%
+%% Examples:
+%% +-------------------+-----------------------------------------------+
+%% | Path              | Meaning                                       |
+%% +-------------------+-----------------------------------------------+
+%% | []                | Depth 0 (No edges)                            |
+%% | [[]]              | Depth 1, no filter (default)                  |
+%% | [[], []]          | Depth 2, no filter                            |
+%% | [[], [depiction]] | Depth 2, only depiction on second order edges |
+%% +-------------------+-----------------------------------------------+
+
+with_deep_edges(Rsc, [], _Context) ->
+    Rsc;
+with_deep_edges(Rsc = #{<<"id">> := Id}, [[]|DeeperPredicates], Context) ->
+    Edges = lists:flatmap(
+              fun({Predicate, PredicateEdges}) ->
+                      [
+                       #{
+                        <<"predicate_name">> => Predicate,
+                        <<"resource">> =>
+                            with_deep_edges(
+                              m_ginger_rest:rsc(
+                                proplists:get_value(object_id, Edge),
+                                Context),
+                              DeeperPredicates, Context)
+                       } || Edge <- lists:reverse(PredicateEdges),
+                            m_rsc:is_visible(proplists:get_value(object_id, Edge), Context)
+                      ]
+              end,
+              m_edge:get_edges(Id, Context)),
+    Rsc#{<<"edges">> => Edges};
+with_deep_edges(Rsc = #{<<"id">> := Id}, [CurrentPredicates|DeeperPredicates], Context) ->
+    Edges = lists:flatmap(
+            fun(Predicate) ->
+                    #rsc_list{list = Objects} = m_rsc:o(Id, Predicate, Context),
+                    [
+                     #{
+                      <<"predicate_name">> => Predicate,
+                      <<"resource">> =>
+                          with_deep_edges(
+                            m_ginger_rest:rsc(Object, Context),
+                            DeeperPredicates,
+                            Context)
+                     } || Object <- Objects
```

Comment:
Ah we don't need the `is_visible` here, because `m_rsc:o` is covering that?

Reply:
Good question, I have a hard time tracing back if that is the case. This code was partly inherited from [m_ginger_rest.erl's with_edges/3](https://github.com/driebit/ginger/blob/master/modules/mod_ginger_base/models/m_ginger_rest.erl#L58) which uses `m_rsc:o` instead of `m_edge:get_edges` in `with_edges/2`.

Reply:
I agree. I'm having a hard time finding that check in `m_rsc:o` too. It seems to just forward to `m_edge:get_edges`. So we actually should check visibility here explicitly too.

Reply:
Great!